libs/auth/src/dipped_smart_card_auth.ts

```diff
@@ -226,6 +234,17 @@ export class DippedSmartCardAuth implements DippedSmartCardAuthApi {
     await this.updateAuthStatus(machineState, { type: 'log_out' });
   }
 
+  async updateSessionExpiry(
```

Comment:
This method exists for two cases:

1. Bringing the session expiry up when the inactive session time limit is hit. I know I could just have the frontend hit the logout endpoint instead of modifying the auth status expiry, but updating the expiry makes accessing the new logout time easier across the UI, which further makes implementing the nav bar timer much easier.
2. Handling the case that the system clock is updated and the expiry needs to be adjusted accordingly. I haven't handled that in this PR but will in a follow-up. (Thanks to @eventualbuddha for reminding me about system clock updating.)

After I've incorporated 2, I may add some restrictions to how the session expiry can be updated for enhanced security such that, even if someone compromises the frontend, they can't extend a session forever.